status/projects.py

```diff
@@ -319,35 +334,35 @@ def list_projects(self, filter_projects='all'):
 
         final_projects = OrderedDict()
         for row in filtered_projects:
-            a=type(row)
             row = self.project_summary_data(row)
-            final_projects[row.key[1]] = row.value
+            proj_id = row.key[1]
+
+            final_projects[proj_id] = row.value
+            for date_type, date in row.value['summary_dates'].items():
+                final_projects[proj_id][date_type] = date
+
             for key, value in def_dates_gen.items():
                 start_date = value[0]
                 end_date = value[1]
-                final_projects[row.key[1]][key] = self._calculate_days_in_status(final_projects[row.key[1]].get(start_date),
-                                                                                    final_projects[row.key[1]].get(end_date))
-
-        # Include dates for each project:
-        for row in self.application.projects_db.view("project/summary_dates", descending=True, group_level=1):
-            if row.key[0] in final_projects:
-                for date_type, date in row.value.items():
-                    final_projects[row.key[0]][date_type] = date
-
-                for key, value in def_dates_summary.items():
-                    if key == 'days_seq_start':
-                        if 'Library, By user' in final_projects[row.key[0]].get('library_construction_method', '-'):
-                             start_date = value[0][1]
-                        else:
-                            start_date = value[0][0]
-                    else:
-                        start_date = value[0]
-                    end_date = value[1]
-                    if key in ['days_prep', 'days_prep_start'] and 'Library, By user' in final_projects[row.key[0]].get('library_construction_method', '-'):
-                        final_projects[row.key[0]][key] = '-'
+                final_projects[proj_id][key] = self._calculate_days_in_status(final_projects[proj_id].get(start_date),
+                                                                                    final_projects[proj_id].get(end_date))
+
+            for key, value in def_dates_summary.items():
```

Comment:
We could probably merge this with the loop above, since we get summary dates in the same view. But its not an immediate need.